mopidy_spotify/web.py

```diff
@@ -128,11 +154,16 @@ def _request_with_retries(self, method, url, *args, **kwargs):
                 logger.debug('Fetching %s failed: %s', prepared_request.url, e)
                 status_code = None
                 backoff_time = None
+                expires = None
                 result = None
             else:
                 status_code = response.status_code
                 backoff_time = self._parse_retry_after(response)
-                result = self._decode(response)
+                expires = self._parse_cache_control(response)
+                etag = self._parse_etag(response)
+                json = self._decode(response)
+                result = WebResponse(
+                    prepared_request.url, json, expires, etag, status_code)
```

Comment:
The `_parse_*` methods doesn't use `self` at all, so they can be placed anywhere.

I suggest keeping `WebResponse.__init__()` as is, but add a classmethod `WebResponse.from_requests(request, response)` that does parsing, etc before calling `__init__()` to create the object.

That change alone moves quite a lot of HTTP code from `OAuthClient` to `WebResponse`.

Reply:
Good idea, done.